components/brave_wallet_ui/common/hooks/use-pending-transaction.ts

```diff
@@ -6,96 +6,280 @@
 
 // Custom Hooks
-import { useTransactionParser } from './transaction-parser'
 import usePricing from './pricing'
 import useTokenInfo from './token'
 import { useLib } from './useLib'
 import {
+  useSafeUISelector,
   useSafeWalletSelector,
   useUnsafeWalletSelector
 } from './use-safe-selector'
-import { useGetNetworkQuery } from '../slices/api.slice'
+import {
+  useGetAccountTokenCurrentBalanceQuery,
+  useGetGasEstimation1559Query,
+  useGetNetworkQuery,
+  useGetSolanaEstimatedFeeQuery,
+  walletApi
+} from '../slices/api.slice'
+import {
+  useAccountQuery,
+  usePendingTransactionsQuery
+} from '../slices/api.slice.extra'
 
 // Constants
 import { BraveWallet } from '../../constants/types'
 import {
   UpdateUnapprovedTransactionGasFieldsType,
   UpdateUnapprovedTransactionNonceType
 } from '../constants/action_types'
-import { isSolanaTransaction, sortTransactionByDate } from '../../utils/tx-utils'
 import { MAX_UINT256 } from '../constants/magics'
 
 export const usePendingTransactions = () => {
   // redux
-  const dispatch = useDispatch()
+  const dispatch = useDispatch<ThunkDispatch<any, any, any>>()
   const accounts = useUnsafeWalletSelector(WalletSelectors.accounts)
-  const transactions = useUnsafeWalletSelector(WalletSelectors.transactions)
-  const transactionInfo = useUnsafeWalletSelector(
-    WalletSelectors.selectedPendingTransaction
+  const selectedPendingTransactionId = useSafeUISelector(
+    UISelectors.selectedPendingTransactionId
   )
   const visibleTokens = useUnsafeWalletSelector(
     WalletSelectors.userVisibleTokensInfo
   )
   const transactionSpotPrices = useUnsafeWalletSelector(
     WalletSelectors.transactionSpotPrices
   )
-  const gasEstimates = useUnsafeWalletSelector(WalletSelectors.gasEstimates)
   const fullTokenList = useUnsafeWalletSelector(WalletSelectors.fullTokenList)
-  const pendingTransactions = useUnsafeWalletSelector(
-    WalletSelectors.pendingTransactions
+  const spotPrices = useUnsafeWalletSelector(
+    WalletSelectors.transactionSpotPrices
   )
   const hasFeeEstimatesError = useSafeWalletSelector(
     WalletSelectors.hasFeeEstimatesError
   )
 
   // queries
+  const { pendingTransactions } = usePendingTransactionsQuery({
+    address: null,
+    chainId: null,
+    coinType: null
+  })
+
+  const transactionInfo = React.useMemo(() => {
+    if (!pendingTransactions.length) {
+      return undefined
+    }
+    return (
+      pendingTransactions.find(
+        (tx) => tx.id === selectedPendingTransactionId
+      ) ?? pendingTransactions[0]
+    )
+    }, [pendingTransactions, selectedPendingTransactionId])
+
+  const txCoinType = transactionInfo
+    ? getCoinFromTxDataUnion(transactionInfo.txDataUnion)
+    : undefined
+
   const { data: transactionsNetwork } = useGetNetworkQuery(
-    transactionInfo
+    transactionInfo && txCoinType !== undefined
+      ? {
+          chainId: transactionInfo.chainId,
+          coin: txCoinType
+        }
+      : undefined
```

Comment:
Probably worth skipping in case of undefined, to avoid cascading call to fetch networks registry.

Reply:
updated to skip